### PR TITLE
docs(deprecations): add deprecation page

### PR DESCRIPTION
**Pull request by JWO719**
This PR is related to #3852

That's how it looks like:
image

I faced some problems with members or overloaded functions that are deprecated. For now we don't even mention this in the dedicated doc page. I think changing this and adding it to the deprecation page could maybe done in an additional pr. But maybe we want to discuss that in detail.

Additionally I changed some of the methods that were marked as deprecated, but just because they were for internal usage, as this. This makes it not be visible in the documentation. That seems to be more logical for me. 

Generally the deprecation text could be improved.

### CODE REVIEW

File: docs_app/tools/transforms/angular-api-package/processors/generateDeprecationsListDoc.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    outputFolder: null,
+    $process: function(docs) {
+      docs.push({
+        docType: 'deprecation',
+        template: 'json-doc.template.json',
+        path: this.outputFolder + '/deprecations',
+        outputPath: this.outputFolder + '/deprecations.json',
+        data: docs
+          .filter(doc => {
+            console.log(doc.members);
```
Comment (benlesh) [inline docs_app/tools/transforms/angular-api-package/processors/generateDeprecationsListDoc.js]: left over console.log

File: src/internal/AsyncSubject.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export class AsyncSubject<T> extends Subject<T> {
   private hasNext: boolean = false;
   private hasCompleted: boolean = false;
 
-  /** @deprecated This is an internal implementation detail, do not use. */
+  /** @internal This is an internal implementation detail, do not use. */
```
Comment (benlesh) [inline src/internal/AsyncSubject.ts]: I feel like we moved from internal to deprecated for some reason related to closure compiler or the like... cc @jasonaden

Comment (rkirov) [reply src/internal/AsyncSubject.ts]: TypeScript --stripInternal will remove @internal properties from the .d.ts which can affect downstream typechecking.

https://www.typescriptlang.org/docs/handbook/compiler-options.html

Comment (JWO719) [reply src/internal/AsyncSubject.ts]: ha interesting. Didn't knew about that. I'll fix that. Thanks for elaborating

**Pull request merged**